components/admission-webhook/README.md

```diff
@@ -0,0 +1,89 @@
+## Goal
+We need a way to inject common data (env vars, volumes) to pods (e.g. notebooks).
+See [issue](https://github.com/kubeflow/kubeflow/issues/2641).
+K8s has [PodPreset](https://kubernetes.io/docs/concepts/workloads/pods/podpreset/) resource with similar use-case, however it is in alpha. 
+K8s admission-controller](https://godoc.org/k8s.io/api/admissionregistration/v1beta1#MutatingWebhookConfiguration) and CRD can be used to implement PodPreset as done in [here](https://github.com/jpeeler/podpreset-crd).
+We borrowed this PodPreset implementation and  customize it for Kubeflow.  
+The code is not directly used as Kubeflow's use case for PodPreset controller is slightly different. 
+In fact, PodPreset in Kubeflow is defined as CRD without the  custom controller (as opposed to [here](https://github.com/jpeeler/podpreset-crd)).
+
+## How this works
+Here is the workflow on how this is used in Kubeflow:
+
+1. Users create  PodPreset manifests which describe additional runtime requirements (i.e., volume, volumeMounts, environment variables) to be injected  into a Pod at creation time.
+PodPresets use label selectors to specify the Pods to which a given PodPreset applies.
+As an example, the following manifest declares a PodPrest to add the secret ```gcp-secret``` in to pods. 
+
+```
+apiVersion: "kubeflow.org/v1alpha1"
+kind: PodPreset
+metadata:
```

Comment:
is podpreset namespaced or cluster scope?

Reply:
podpreset is a crd which is cluster scope by default.

Reply:
It's namespaced right?
see kubeflow/admission-webhook/podpreset-crd.libsonnet 
What do you mean "by default"?

Reply:
Can you put some namespace in the spec, and mention that each user will only view and apply the podpreset in the namespace?

Reply:
I meant by default, CRD itself is  non-namespaced and is available to all namespaces (https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/)
 Sure will added scope in the spec.